src/JustSaying/AwsTools/MessageHandling/Dispatch/MessageDispatcher.cs

```diff
@@ -175,34 +175,41 @@ private async Task<bool> CallMessageHandler(string queueName, Message message, C
 
             using (_messagingMonitor.MeasureDispatch())
             {
+                var context = new HandleMessageContext(message, messageType, queueName);
                 bool dispatchSuccessful = false;
                 try
                 {
-                    var context = new HandleMessageContext(message, messageType, queueName);
                     dispatchSuccessful = await middleware.RunAsync(context, null, cancellationToken)
                         .ConfigureAwait(false);
                 }
                 finally
                 {
                     watch.Stop();
 
-                    var logMessage =
-                        "{Status} handling message with Id '{MessageId}' of type {MessageType} in {TimeToHandle}ms.";
-                    if (dispatchSuccessful)
+                    using (_logger.BeginScope(new[]
                     {
-                        _logger.LogInformation(logMessage,
-                            "Succeeded",
-                            message.Id,
-                            messageType,
-                            watch.ElapsedMilliseconds);
-                    }
-                    else
+                        new KeyValuePair<string, object>("MessageSource", context.QueueName),
+                        new KeyValuePair<string, object>("SourceType", "Queue")
```

Comment:
Would `KeyValuePair.Create(..., ...)` work here and the co(contra?)-variance would make it compile with it wanting the second type parameter needing to be an object and not the actual value's type, then it's a little less verbose. No worries if not.

Reply:
It wouldn't be able to pick a best inferred type from experiments without giving the array a type. We don't have `KeyValuePair.Create` unfortunately anyway as it's a netstandard2.1 thing

Reply:
The [Serilog.AspNetCore docs](https://github.com/serilog/serilog-aspnetcore#pushing-properties-to-the-iloggert) show this pattern instead, which is a bit nicer:

```csharp
// Microsoft.Extensions.Logging ILogger<T>
// Yes, it's required to use a dictionary. See https://nblumhardt.com/2016/11/ilogger-beginscope/
using (logger.BeginScope(new Dictionary<string, object>
{
    ["UserId"] = "svrooij",
    ["OperationType"] = "update",
}))
{
   // UserId and OperationType are set for all logging events in these brackets
}
```